### ISSUE TITLE
Watcher using a policy that shouldn't hit the network, can still hit the network

### ISSUE THREAD

**Issue by potrebic** [apollographql/apollo-ios#2170]
## Bug report

Via code inspection of `GraphQLQueryWatcher` one can see calling ApolloClient.watch with a cache policy of `.returnCacheDataDontFetch` that there are code paths that will fetch the query from remote.

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 0.50.0
- Xcode version: 13.*
- Swift version: 5.*
- Package manager: *

## Steps to reproduce

Code inspection issue.

## Further details

In the store() method in GraphQLQueryWatcher.swift the following block of code shows the issue, where it triggers a fetch of the query from the server.
```
        case .failure:
          // If the cache fetch is not successful, for instance if the data is missing, refresh from the server.
          self.fetch(cachePolicy: .fetchIgnoringCacheData)
        }
```

If the cache policy given to the watch(query:cachePolicy:) method of the ApolloClient was `.returnCacheDataDontFetch` then I don't think it's proper for the watcher to none the less trigger a remote fetch of the query.

**Comment by AnthonyMDev** [comment]
Haha yeah... I recall being made aware of this issue not too long ago. I thought we had resolved it though! Somehow this must have fallen through the cracks. I'll look into a fix now and try to get it in for the next version release, which is coming soon!

**Comment by AnthonyMDev** [comment]
Okay, fix is up and should be merged in for the 0.51 version we will be releasing in the next few days. Thanks for bringing this  (back) to our attention!
